src/ddmd/dclass.d

```diff
@@ -827,9 +828,21 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
      */
     final bool isAbstract()
     {
+        enum log = false;
         if (isabstract != ABSfwdref)
             return isabstract == ABSyes;
 
+        if (log) printf("isAbstract(%s)\n", toChars());
```

Comment:
Shouldn't this be in `version(none)` or part of some global level `LOG` mechanism?

Perhaps a new bugzilla issue is better place to discuss thoughts though. As debug tracing the compiler internals in a consistent manner would probably require a new logging interface.

Reply:
I've used various logging schemes,  and I like this one the best. Global schemes produce a blizzard of noise and the signal gets lost. This one is very local, and since `log` is an enum it gets optimized away to nothing when `false`.

Reply:
Each module could define its own subsystem, then you turn them on via the command line.